Fix cross-origin error for Chrome

sainthkh opened the pull request:
- Closes #5270

### User facing changelog

Fix cross-origin error for Chrome

### Additional details

- Why was this change necessary?

Firefox cross-origin error was solved. But it was not in Chrome.

- What is affected by this change?

N/A

- Any implementation details to explain?

The fix is targeted to Firefox branch because the issue isn't urgent and almost everything is solved in there. 

Chrome doesn't return `[object Window]` even if it is a Window object. So, we check the error stack message. 

### How has the user experience changed?

It doesn't fail even if cross-origin script is added in Chrome.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue been tagged with a release in ZenHub? <!-- (internal team only)-->
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [NA] Have API changes been updated in the [`type definitions`](cli/types/index.d.ts)?
- [NA] Have new configuration options been added to the [`cypress.schema.json`](cli/schema/cypress.schema.json)?

sainthkh commented:
The failures here are caused by #1096, right?

sainthkh commented:
Rebased to develop branch. Everything clean and clear.

Bkucera commented on packages/driver/test/cypress/fixtures/with_stripe.html:
```diff
@@ -0,0 +1,7 @@
+<!DOCTYPE html>
+<head>
+  <script src="https://js.stripe.com/v3"></script>
```
we can change this to localhost:3501 so we're not relying on third party code

sainthkh commented:
1. Changed the the url to `http://localhost:3501/fixtures/generic.html`.
2. The cause of the failure wasn't Chrome always returning `[object Object]`. When an <iframe> tag has `name` attribute, Firefox returns `[object Window]`, but Chrome returns `[object Object]`. So, I updated the comment. 
3. After updating the url to `generic.html`, I found something weird. `fixtures/generic.html` does have an element with `foo` class and it is clearly visible. But as it is from cross-origin source, Cypress cannot access it. So, the test passes. Maybe, we need to tell the users that Cypress cannot detect elements in cross-origin source iframe. (I think [this](https://docs.cypress.io/faq/questions/using-cypress-faq.html#How-do-I-test-elements-inside-an-iframe) is the good place to add this. Maybe adding link to [this document](https://docs.cypress.io/guides/guides/web-security.html#Limitations), too.)

sainthkh commented:
Failed because of a flaky test.

jennifer-shehane commented:
@Bkucera @flotwig Any chance to review this?

Bkucera commented on packages/driver/src/cy/chai.js:
```diff
@@ -97,6 +97,18 @@ chai.use((chai, u) => {
     if (val && (getType(val) === 'Window')) {
       return '[window]'
     }
```
can you replace this logic with the try-catch as well? I think the try-catch is a better strategy

sainthkh replied:
I realized that this `if` statement can be removed entirely and try-catch can handle it, too. So, I removed it and `getType`.

sainthkh commented:
* Rebased and applied review. 
* Flaky failures.
